New 4.7 Quick Starts focusing on the developer experience

serenamarie125 opened the pull request:
Adding five (5) new Quick Starts for 4.7, which focus on the developer experience.

4 of these are available for all users:
- Get started with Node
- Get started with Quarkus with a Helm Chart
- Get started with Quarkus with s2i
- Get started with Spring Boot

1 needs special permissions:
- Manage available content in the Helm Chart Catalog

Note: Each of these have specific icons
I've attached a screenshot of Quick Starts page with them all installed:

<img width="1792" alt="4 7QuickStart-MoreForTheDeveloper" src="https://user-images.githubusercontent.com/12504403/105916665-4858a400-5fff-11eb-9188-58b0845eea6c.png">

Link to videos:

- [Get started with Node](https://drive.google.com/file/d/1FqQHc1yBVdi9inxHm0zfd2-BrHOUnQGb/view?usp=sharing) 
- [Get started with Quarkus using a Helm Chart](https://drive.google.com/file/d/1mhNAxlSSeqDaJWo941lyH3BDjEHJuFKi/view?usp=sharing)
- [Get started with Quarkus with s2i](https://drive.google.com/file/d/1PUtEiMoPuX-z78D11RLB0NIItpTGrux2/view?usp=sharing)

serenamarie125 commented:
FYI @christianvogt @abhinandan13jan @andrewballantyne I'll add some videos in a bit.  LMK if there's anything else I need to get this prioritized for review for 4.7!  Thanks

serenamarie125 commented:
@beaumorley please review !

serenamarie125 commented:
@maxandersen I know you already reviewed the Quarkus Quick Starts on the test cluster today, could you comment here for awareness?  I addressed all issues but the "name" of the Helm Chart.  If that needs to be changed, we have to get Karena to make that change first, and we can update in a follow up

serenamarie125 commented:
@Preeticp these Quick Starts are aligned with dev use cases, would you mind taking a quick review as well ?

beaumorley commented:
Ran across a few issues with the Quarkus with a Helm Chart QS. I looked at this using cluster-bot and release 4.7.0-0.nightly-2021-01-26-225405. I loaded the Quick Starts with ConsoleQuickStart.

When I click on the Quarkus v0.0.3 card from the Helm Catalog the side panel covered the Quick Start panel.

![image](https://user-images.githubusercontent.com/52013984/105928944-86ac8e00-6014-11eb-90b8-cbc448637474.png)

Also interesting that when a Helm Chart is added the side panel is automatically opened (I believe this is by design). Unfortunately, this hides the Quarkus HR. You have to scroll or close the side panel to see it.

![image](https://user-images.githubusercontent.com/52013984/105929321-54e7f700-6015-11eb-8350-bc27918bbae2.png)

Also I ran across a formatting issue in Step 3:

![image](https://user-images.githubusercontent.com/52013984/105929394-7943d380-6015-11eb-8237-4e38313dff49.png)

Step 4 is duplicated.

![image](https://user-images.githubusercontent.com/52013984/105929688-23bbf680-6016-11eb-9b54-73cd488136ee.png)

andrewballantyne commented:
> FYI @christianvogt @abhinandan13jan @andrewballantyne I'll add some videos in a bit. LMK if there's anything else I need to get this prioritized for review for 4.7!
> 
> FYI @sspeiche @angellk

Get content sign offs from the needed stakeholders and we should be good to go @serenamarie125.

@spadgett Would you mind tossing an `approve` label on this? We can address the LGTM when the content is verified and good to go.
/hold

I'll hold the PR so it won't accidentally merge in the meantime.

serenamarie125 commented:
@jamesfalkner please provide review !

metacosm commented on quickstarts/springboot-with-s2i.yaml:
```diff
@@ -0,0 +1,186 @@
+  icon: >-
+    data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGlkPSJMYXllcl8xIiBkYXRhLW5hbWU9IkxheWVyIDEiIHZpZXdCb3g9IjAgMCAxMDI0IDEwMjQiPjxkZWZzPjxzdHlsZT4uY2xzLTF7ZmlsbDojMTUzZDNjO30uY2xzLTJ7ZmlsbDojZDhkYTlkO30uY2xzLTN7ZmlsbDojNThjMGE4O30uY2xzLTR7ZmlsbDojZmZmO30uY2xzLTV7ZmlsbDojM2Q5MTkxO308L3N0eWxlPjwvZGVmcz48dGl0bGU+c25vd2Ryb3BfaWNvbl9yZ2JfZGVmYXVsdDwvdGl0bGU+PHBhdGggY2xhc ...
+  introduction: >-
+    **Spring** is a Java framework for building applications based on a distributed microservices architecture. 
```
```suggestion
    **Spring Boot** is a Java framework for building applications based on a distributed microservices architecture. 
```
Also, you might want to check requirements when it comes to trademarks…

jamesfalkner commented:
> SpringBoot QS:
> 
> In the SpringBoot QS the icon on the card is incorrect.

Unfortunately, we can't use the upstream spring logo(s) due to [commercial use trademark policy](https://spring.io/trademarks).

This logo is from the Red Hat-led [Snowdrop community](https://github.com/snowdrop).

jamesfalkner commented on quickstarts/quarkus-with-helm.yaml:
```diff
@@ -0,0 +1,162 @@
+  conclusion: >-
+    Your Quarkus application is deployed and ready!
+  description: 'Deploy a Quarkus Application from a Helm Chart.'
+  displayName: Get started with Quarkus with a Helm Chart
```
If the phrase "Get started _with_ X _with_ a Helm Chart" is standard, then cool, but the duplication triggered me and I think it may be better to be "Get started with Quarkus _using_ a Helm Chart"

jamesfalkner commented on quickstarts/quarkus-with-helm.yaml:
```diff
@@ -0,0 +1,162 @@
+        success: Your deployment has one running pod.
+      title: View the pod status
+    - description: >-
+        Now let's Open the Quarkus application URL!  
```
Open --> open ?

serenamarie125 replied:
I'm going to change this sentence across the board.

jamesfalkner commented on quickstarts/quarkus-with-s2i.yaml:
```diff
@@ -0,0 +1,198 @@
+        success: You changed the icon associated with your deployment.
+      title: Change the deployment icon to Quarkus
+    - description: >-
+        Now let's Open the Quarkus application URL!  
```
Open --> open?

serenamarie125 replied:
I'm going to change this sentence across the board.

jamesfalkner commented:
> @jamesfalkner please provide review !

Done! I focused on the content and words and made a few suggestions.

serenamarie125 commented:
> When I click on the Quarkus v0.0.3 card from the Helm Catalog the side panel covered the Quick Start panel.
@beaumorley https://issues.redhat.com/browse/ODC-5400 was entered to capture this issue.  This is a Quick Start issue, not specific to the quick start content which is part of this PR

serenamarie125 commented:
@beaumorley 

> Also I ran across a formatting issue in Step 3:

This is fixed

> Step 4 is duplicated.

This is fixed

> In the SpringBoot QS the icon on the card is incorrect.

We are using the rh-spring-boot ( upstream snowdrop ) icon per the team's request

> When I added From Git I did not change the application name because the instructions did not say to. I had already added something previously so my Spring Boot deployment got added to an existing Application. So the check your work was wrong. We may want to specify to have them create the application name.

I changed all of these to tell the user to create a new project, thus this no longer happens.

serenamarie125 commented:
@metacosm @jamesfalkner @beaumorley all comments have been addressed!

angellk commented:
lgtm

beaumorley commented:
lgtm!

jamesfalkner commented:
lgtm!

Preeticp commented on quickstarts/quarkus-with-helm.yaml:
```diff
@@ -0,0 +1,138 @@
+  introduction: >-
+    **Quarkus** is a Cloud Native, (Linux) Container First framework for writing Java applications.
+
+    - **Container First**: Minimal footprint Java applications optimal for running in containers.
```
...applications that are optimal...

serenamarie125 commented:
FYI @openshift/quickstart-ux-content

Preeticp commented on quickstarts/quarkus-with-helm.yaml:
```diff
@@ -0,0 +1,138 @@
+
+    - **Standards-based**: Based on the standards and frameworks you love and use (RESTEasy and JAX-RS, Hibernate ORM and JPA, Netty, Eclipse Vert.x, Eclipse MicroProfile, Apache Camel...).
+
+    - **A great choice for microservices and serverless**: Brings lightning fast startup time and code turn around to Java apps.
```
turnaround

Preeticp commented on quickstarts/quarkus-with-helm.yaml:
```diff
@@ -0,0 +1,138 @@
+        
+            - The form includes multiple sections which are collapsed by default.  Browse these sections to get an idea of the available options, although no changes are needed to proceed. 
+        
+        1. At the bottom of the form, click **Install** to create the Quarkus application via the Helm Chart.
```
suggest - via/using

Preeticp commented on quickstarts/quarkus-with-helm.yaml:
```diff
@@ -0,0 +1,138 @@
+        
+        1. Click on the icon to open the URL and run the application.  The application will be run in a new browser tab.
+
+        Now let's Open the Quarkus application URL!  
```
open - small o

Preeticp commented on quickstarts/quarkus-with-s2i.yaml:
```diff
@@ -0,0 +1,210 @@
+  name: quarkus-with-s2i
+spec:
+  conclusion: Your Quarkus application is deployed and ready!
+  description: 'Import a Quarkus Application from git, build and deploy it onto OpenShift.'
```
comma after build

Preeticp commented on quickstarts/quarkus-with-s2i.yaml:
```diff
@@ -0,0 +1,210 @@
+    #### Quarkus is a Cloud Native, (Linux) Container First framework for
+    writing Java applications.
+
+    - **Container First:** Minimal footprint Java applications optimal for
```
that are optimal

Preeticp commented on quickstarts/quarkus-with-helm.yaml:
```diff
@@ -0,0 +1,138 @@
+
+    - **Unify imperative and reactive**: Brings under one programming model non-blocking and imperative styles of development.
+
+    - **Standards-based**: Based on the standards and frameworks you love and use (RESTEasy and JAX-RS, Hibernate ORM and JPA, Netty, Eclipse Vert.x, Eclipse MicroProfile, Apache Camel...).
```
I am not sure why we are using parenthesis here?
Suggest removing the parenthesis as well as the ellipsis after Apache Camel

Preeticp commented on quickstarts/quarkus-with-s2i.yaml:
```diff
@@ -0,0 +1,210 @@
+    non-blocking and imperative styles of development.
+
+    - **Standards-based**: Based on the standards and frameworks you love and
+    use (RESTEasy and JAX-RS, Hibernate ORM and JPA, Netty, Eclipse Vert.x,
```
Suggest removing the parenthesis as well as the ellipsis after Apache Camel

Preeticp commented on quickstarts/quarkus-with-s2i.yaml:
```diff
@@ -0,0 +1,210 @@
+    Eclipse MicroProfile, Apache Camel...).
+
+    - **A great choice for microservices and serverless**: Brings lightning fast startup
+    time and code turn around to Java apps.
```
turnaround

Preeticp commented on quickstarts/quarkus-with-s2i.yaml:
```diff
@@ -0,0 +1,210 @@
+    - **A great choice for microservices and serverless**: Brings lightning fast startup
+    time and code turn around to Java apps.
+
+    - **Developer Joy**: Development centric experience without compromise to
```
How about: Development centric experience bringing your amazing apps to life in no time.
There is another place above which uses the same line above.

Preeticp commented on quickstarts/quarkus-with-s2i.yaml:
```diff
@@ -0,0 +1,210 @@
+        
+        1. The icon on the bottom right
+        quadrant of the **quarkus-quickstarts** deployment represents the Git
+        repo off the associated code.  The icon shown can be for Bitbucket,
```
repo/repository

Preeticp commented on quickstarts/quarkus-with-s2i.yaml:
```diff
@@ -0,0 +1,210 @@
+        associated with the quarkus-quickstarts deployment to indicate that it
+        is a Quarkus application. 
+
+        1. Click on the **quarkus-quickstarts** deployment 
```
missing period `.`

afialkoff replied:
```suggestion
        1. Click on the **quarkus-quickstarts** deployment. 
```

Preeticp commented on quickstarts/quarkus-with-s2i.yaml:
```diff
@@ -0,0 +1,210 @@
+        1. In the side panel, click on the **Actions** drop-down and click the
+        **Edit labels** menu item. 
+
+        1. Click on the “x” next to the `app.openshift.io/runtime=java` label to
```
missing period .

Preeticp commented on quickstarts/quarkus-with-s2i.yaml:
```diff
@@ -0,0 +1,210 @@
+        1. Click on the “x” next to the `app.openshift.io/runtime=java` label to
+        remove the entry 
+
+        1. Add an additional label 
```
colon`:` after `label`

Preeticp commented on quickstarts/quarkus-with-s2i.yaml:
````diff
@@ -0,0 +1,210 @@
+          app.openshift.io/runtime=quarkus
+          ```
+
+        1. Click on **Save** to save the labels
````
missing period .

Preeticp commented on quickstarts/quarkus-with-s2i.yaml:
```diff
@@ -0,0 +1,210 @@
+        
+        1. Click on the icon to open the URL and run the application.  The application will be run in a new browser tab.
+
+        Now let's Open the Quarkus application URL!  
```
open

Preeticp commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+spec:
+  conclusion: >-
+    Your Spring application is deployed and ready! 
+  description: 'Import a Spring Application from git, build and deploy it onto OpenShift.'
```
comma after `build`

Preeticp commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+  icon: >-
+    data:image/svg+xml;base64,PHN2ZyB4bWxucz0iaHR0cDovL3d3dy53My5vcmcvMjAwMC9zdmciIGlkPSJMYXllcl8xIiBkYXRhLW5hbWU9IkxheWVyIDEiIHZpZXdCb3g9IjAgMCAxMDI0IDEwMjQiPjxkZWZzPjxzdHlsZT4uY2xzLTF7ZmlsbDojMTUzZDNjO30uY2xzLTJ7ZmlsbDojZDhkYTlkO30uY2xzLTN7ZmlsbDojNThjMGE4O30uY2xzLTR7ZmlsbDojZmZmO30uY2xzLTV7ZmlsbDojM2Q5MTkxO308L3N0eWxlPjwvZGVmcz48dGl0bGU+c25vd2Ryb3BfaWNvbl9yZ2JfZGVmYXVsdDwvdGl0bGU+PHBhdGggY2xhc ...
+  introduction: >-
+    **Spring** is a Java framework for building applications based on a distributed microservices architecture. 
+    
+    - Spring enables easy packaging and configuration of Spring applications into a self-contained executable application which can be easily deployed as a container to OpenShift.
+    
+    - Spring applications can integrate OpenShift capabilities to provide a natural "Spring on OpenShift" developer experience for both existing and net-new Spring applications. For example:
+    
+    - Externalized Configuration using Kubernetes ConfigMaps and integration with Spring Cloud Kubernetes
```
configuration

afialkoff replied:
This intro seems like a wealth of information, but how much of it is entirely relevant to this quick start? Can we unify the bulleted list by starting with the same article of speech and really zeroing in on how getting started with Spring is valuable to OpenShift users?

Preeticp commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+    - Load Balancing with Replication Controllers
+    
+    - Kubernetes Health Probes and integration with Spring Actuator
+    
```
discovery; balancing; health probes;

serenamarie125 replied:
updated

Preeticp commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+    
+    - Metrics: Prometheus, Grafana, and integration with Spring Cloud Sleuth
+    
+    - Distributed Tracing with Istio & Jaeger Tracing
```
tracing in both instances?

afialkoff replied:
```suggestion
    - Distributed tracing with Istio & Jaeger tracing
```

Preeticp commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+      title: View the pod status
+
+    - description: >-
+        Since a Java builder image was used during the Import from Git flow, a Java icon is used by default in topology.  
```
**Import from Git** in other instances above too

afialkoff replied:
```suggestion
        Since a Java builder image was used during the **Import from Git** flow, a Java icon is used by default in Topology view.  
```

Preeticp commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+
+          - Let’s change the icon associated with the **rest-http-example** deployment to indicate that it is a Spring application.
+        
+        1. Click on the **rest-http-example** deployment
```
missing period .

afialkoff replied:
```suggestion
        1. Click on the **rest-http-example** deployment.
```

Preeticp commented on quickstarts/node-with-s2i.yaml:
```diff
@@ -0,0 +1,155 @@
+
+        1. Enter the name for your new project in the **Name** field, and click **Create**.
+        
+        1. On the Add page, click **From Git**. 
```
**+Add** page

Preeticp commented on quickstarts/quarkus-with-helm.yaml:
```diff
@@ -0,0 +1,138 @@
+
+        1. Enter the name for your new project in the **Name** field, and click **Create**.
+        
+        1. On the Add page, click **Helm Chart**. 
```
**+Add** page

Preeticp commented on quickstarts/quarkus-with-s2i.yaml:
```diff
@@ -0,0 +1,210 @@
+
+        1. Enter the name for your new project in the **Name** field, and click **Create**.
+        
+        1. On the Add page, click **From Git**. 
```
**+Add**

Preeticp commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+
+        1. Enter the name for your new project in the **Name** field, and click **Create**.
+        
+        1. On the Add page, click **From Git**. 
```
**+Add**

afialkoff replied:
```suggestion
        1. On the **+Add** page, click **From Git**. 
```

Preeticp reviewed (commented):
Minor language and formatting based comments. HTH

geoand commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+  introduction: >-
+    **Spring** is a Java framework for building applications based on a distributed microservices architecture. 
+    
+    - Spring enables easy packaging and configuration of Spring applications into a self-contained executable application which can be easily deployed as a container to OpenShift.
```
`Spring enables easy packaging and configuration of Spring applications` doesn't sound right

lholmquist commented on quickstarts/node-with-s2i.yaml:
```diff
@@ -0,0 +1,155 @@
+        To view the pod status:
+                
+          - Notice that the
+        **nodejs-rest-http-redhat** deployment has a pod donut imposed on the
```
same as the comment above  `nodejs-rest-http-redhat` -> `nodejs-rest-http`

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+    
+    - Externalized Configuration using Kubernetes ConfigMaps and integration with Spring Cloud Kubernetes
+    
+    - Service Discovery using Kubernetes Services
```
```suggestion
    - Service discovery using Kubernetes services
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+    
+    - Load Balancing with Replication Controllers
+    
+    - Kubernetes Health Probes and integration with Spring Actuator
```
```suggestion
    - Kubernetes health probes and integration with Spring Actuator
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+    - description: >-
+        To create a Spring application:
+        
+        1. Click on the [perspective switcher]{{highlight qs-perspective-switcher}} at the top of the navigation, and select **Developer**.
```
```suggestion
        1. Enter the Developer perspective: In the main navigation, click the dropdown menu and select **Developer**.
```

serenamarie125 replied:
sticking with perspective switcher to be consistent with dev console docs

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+        
+        1. Click on the [perspective switcher]{{highlight qs-perspective-switcher}} at the top of the navigation, and select **Developer**.
+        
+        1. In the navigation, click [Add]{{highlight qs-nav-add}}.
```
```suggestion
        1. In the navigation menu, click [+Add]{{highlight qs-nav-add}}.
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+        
+        1. In the navigation, click [Add]{{highlight qs-nav-add}}.
+        
+        1. Using the project drop-down list, select **Create Project** to create a new project for your Spring application.
```
```suggestion
        1. Click the **Project** dropdown menu and select **Create Project**.
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+        
+        1. Using the project drop-down list, select **Create Project** to create a new project for your Spring application.
+
+        1. Enter the name for your new project in the **Name** field, and click **Create**.
```
```suggestion
        1. In the **Name** field, enter a name for your new project. Then click **Create**.
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+        
+        1. On the Add page, click **From Git**. 
+        
+        1. In the form, add 
```
```suggestion
        1. In the **Git Repo URL** field, add: 
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
````diff
@@ -0,0 +1,208 @@
+          ```
+            https://github.com/snowdrop/rest-http-example
+          ``` 
+          to the **Git Repo URL** field.
````
```suggestion

```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
````diff
@@ -0,0 +1,208 @@
+          ``` 
+          to the **Git Repo URL** field.
+        
+        1. Click **Show advanced Git options**, which will expose additional fields.
````
```suggestion
        1. Click **Show advanced Git options** to expose additional form fields.
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
````diff
@@ -0,0 +1,208 @@
+        
+        1. Click **Show advanced Git options**, which will expose additional fields.
+        
+        1. Add 
+        
+          ```
+          2.3.4-2-redhat
+          ``` 
+        to the **Git reference** field.
````
```suggestion
        1. In the **Git reference** field, add:
        
          ```
          2.3.4-2-redhat
          ``` 
       
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
````diff
@@ -0,0 +1,208 @@
+          ``` 
+        to the **Git reference** field.
+        
+        1. At the bottom of the form, click **Create** to create your Spring application. 
````
```suggestion
        1. At the end of the form, click **Create**. 
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+      review:
+        failedTaskHelp: This task isn’t verified yet. Try the task again.
+        instructions: >-
+          The application is represented by the light grey area with the white border.  The deployment is a white circle.  To verify the application was successfully created:
+          
+          1. Do you see a **rest-http-example-app** application?
+          
+          1. Do you see a **rest-http-example** deployment?
```
```suggestion
          The application is represented by the light grey area with the white border. The deployment is represented by a white circle. 
           
          Verify that the application was successfully created:
          
          1. Do you see a **rest-http-example-app** application?
          
          1. Do you see a **rest-http-example** deployment?
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+    - description: >-
+        To view the build status of the Spring application:
+        
+        1. Hover over the icon on the bottom left quadrant of the **rest-http-example** deployment to see the build status in a tooltip.
```
```suggestion
        1. . To view build status in a tooltip, hover over the status icon on the bottom left quadrant of the **rest-http-example** deployment.
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+        
+        1. Hover over the icon on the bottom left quadrant of the **rest-http-example** deployment to see the build status in a tooltip.
+        
+        1. Click on the icon for quick access to the build log.  
```
```suggestion
        1. Click on the status icon to access the build log.  
```

serenamarie125 replied:
concerns with use of status icon here

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+        
+        1. Click on the icon for quick access to the build log.  
+        
+            - You should be able to see the log stream of the **rest-http-example-1** build on the **Build Details** page.
```
```suggestion
            - To view the the log stream of the **rest-http-example-1** build, select the **Details** tab.
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+        failedTaskHelp: >-
+          This task isn’t verified yet. Try the task again.
+        instructions: |-
+          To verify the build is complete:
```
```suggestion
         Verify that the build is complete:
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+        instructions: |-
+          To verify the build is complete:
+
+          - Wait for the build to complete.  It may take a few minutes.  Do you see a **Complete** badge on the page header?
```
```suggestion
          - This build may take a few minutes. When it's finished, a **Complete** badge will surface on the page header beside build name **rest-http-example**. Did this badge appear?
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+    - description: >-
+        To view the associated code:
+        
+        1. In the navigation, click [Topology]{{highlight qs-nav-topology}}.
```
```suggestion
        1. In the navigation menu, click [Topology]{{highlight qs-nav-topology}}.
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+        failedTaskHelp: >-
+          This task isn’t verified yet. Try the task again.
+        instructions: |-
+          Verify you can see the code associated with the sample app:
```
```suggestion
          Verify that you can see the code associated with the sample app:
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+
+      summary:
+        failed: Try the steps again.
+        success: You accessed the git repo associated with your deployment.
```
```suggestion
        success: You accessed the Git repo associated with your deployment.
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+    - description: >-
+        To view the pod status:
+        
+            - Notice that the
+        **rest-http-example** deployment has a pod donut imposed on the
+        circle, representing the pod status (i.e. blue = running).  
+            - The color of the donut indicates
+        the pod status.  
```
```suggestion
            - A pod donut imposed on the **rest-http-example** deployment circle represents the pod status. Pod status is indicated by color.
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+            - The color of the donut indicates
+        the pod status.  
+
+        1. Hover over the pod donut to see the pod status in a tooltip.
```
```suggestion
        1. To view pod status in a tooltip, hover over the pod donut.
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+        failedTaskHelp: >-
+          This task isn’t verified yet. Try the task again.
+        instructions: |-
+          To verify you see the pod status:
```
```suggestion
          Verify that you can view the pod status tooltip:
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+    - description: >-
+        Since a Java builder image was used during the Import from Git flow, a Java icon is used by default in topology.  
+        
+          - This icon is determined by the value of the app.openshift.io/runtime label.
+
+          - Let’s change the icon associated with the **rest-http-example** deployment to indicate that it is a Spring application.
```
```suggestion
          -  The displayed icon is determined by the value of the `app.openshift.io/runtime` label. You can change the icon associated with a deployment to indicate its application type. 
          
          To associate a Spring icon with the **rest-http-example** deployment:
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+        
+        1. Click on the **rest-http-example** deployment
+        
+        1. In the side panel, click on the Actions drop-down and click the Edit labels menu item.
```
```suggestion
        1. In the side panel, click the **Actions** dropdown menu and select **Edit labels**.
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+        
+        1. In the side panel, click on the Actions drop-down and click the Edit labels menu item.
+        
+        1. Click on the “x” next to the **app.openshift.io/runtime=java** label to remove the entry
```
```suggestion
        1. Click on the “x” next to the **app.openshift.io/runtime=java** label to remove it from the labels list.
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+        
+        1. Click on the “x” next to the **app.openshift.io/runtime=java** label to remove the entry
+        
+        1. Add an additional label 
```
```suggestion
        1. Add an additional label: 
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
````diff
@@ -0,0 +1,208 @@
+          app.openshift.io/runtime=rh-spring-boot
+          ```
+        
+        1. Click on **Save** to save the labels
````
```suggestion
        1. Click on **Save** to save your edited labels.
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+        failedTaskHelp: >-
+          This task isn’t verified yet. Try the task again.
+        instructions: |-
+          To verify you have updated the icon :
```
```suggestion
          Verify that the icon has updated successfully:
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+        instructions: |-
+          To verify you have updated the icon :
+          
+          - Do you now see the Spring icon rather than a java icon in the **rest-http-example** deployment? 
```
```suggestion
          - Has the Spring icon replaced the Java icon in the **rest-http-example** deployment? 
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+      title: Change the deployment icon to Spring
+
+    - description: >-
+        Now let's run the Spring application!  
```
```suggestion

```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+    - description: >-
+        Now let's run the Spring application!  
+        
+          - The icon on the top right
+        quadrant of the **rest-http-example** deployment represents the route
+        URL.
+
+        1.  Click on the icon to open the URL and run the application.  
```
```suggestion
          - The external link icon on the top right
        quadrant of the **rest-http-example** deployment represents the route
        URL.

        1.  Click the external link icon to open the URL and run the application in a new browser tab.  
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+
+        1.  Click on the icon to open the URL and run the application.  
+        
+          - The application will be run in a new browser tab.
```
```suggestion

```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+        failedTaskHelp: >-
+          This task isn’t verified yet. Try the task again.
+        instructions: |-
+          To verify your Spring application is running:
```
```suggestion
          Verify that your Spring application is running:
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+        instructions: |-
+          To verify your Spring application is running:
+
+          - Does the new tab have a **HTTP Example** title?
```
```suggestion
          - Is the new tab titled **HTTP Example**?
```

afialkoff commented on quickstarts/spring-with-s2i.yaml:
```diff
@@ -0,0 +1,208 @@
+
+      summary:
+        failed: Try the steps again.
+        success: Your Spring application is running!
```
```suggestion
        success: Your Spring application is running.
```

afialkoff reviewed (changes requested):
Hi @serenamarie125! In this review, you'll find some general copy edits and suggestions along with a few questions about two of the quick start introductions.

Let me know if you have any questions. I hope you find this feedback useful, and I look forward to working with you more in the future! 🙂

serenamarie125 commented on quickstarts/manage-helm-repos.yaml:
```diff
@@ -0,0 +1,79 @@
+          
+          1. Go to the Helm Chart Catalog
+          
+          1. Confirm that the **azure-sample-repo** is one of the available catalogs.
```
this is adding it as an additional catalog, so additional helm charts are exposed from that catalog.  I changed to:
Confirm that the **azure-sample-repo** is an available catalog

serenamarie125 commented on quickstarts/manage-helm-repos.yaml:
```diff
@@ -0,0 +1,79 @@
+    - description: >-
+        Follow these steps to create a HelmChartRepository CR:
+
+        1. Click on the [perspective switcher]{{highlight qs-perspective-switcher}} at the top of the navigation, and select **Administrator**.
```
Keeping perspective switcher for now to be consistent with Dev console docs

serenamarie125 commented on quickstarts/node-with-s2i.yaml:
```diff
@@ -0,0 +1,155 @@
+    - description: >-
+        To create a Node application:
+        
+        1. Click on the [perspective switcher]{{highlight qs-perspective-switcher}} at the top of the navigation, and select **Developer**.
```
Keeping perspective switcher for now to be consistent with Dev console docs

serenamarie125 commented on quickstarts/manage-helm-repos.yaml:
```diff
@@ -0,0 +1,79 @@
+        
+        1. In the navigation, click [Home]{{highlight qs-nav-home}}.
+        
+        1. In the [Home navigation section]{{highlight qs-nav-home}}, click **Search**.
```
didn't take this suggestion, sticking with perspective switcher to be consistent with dev console docs

serenamarie125 commented on quickstarts/node-with-s2i.yaml:
````diff
@@ -0,0 +1,155 @@
+        1. In the form, add 
+        
+          ```
+            https://github.com/nodeshift-starters/nodejs-rest-http-redhat
````
worked with @jamesfalkner on determining which repo to use.

serenamarie125 commented on quickstarts/node-with-s2i.yaml:
```diff
@@ -0,0 +1,155 @@
+        
+        1. Hover over the icon on the bottom left quadrant of the **nodejs-rest-http-redhat** deployment to see the build status in a tooltip.
+        
+        1. Click on the icon for quick access to the build log.  
```
This icon is associated with the build, so I don't think saying status icon makes sense

serenamarie125 commented on quickstarts/quarkus-with-helm.yaml:
```diff
@@ -0,0 +1,138 @@
+    - description: >-
+        To create a Quarkus application:
+        
+        1. Click on the [perspective switcher]{{highlight qs-perspective-switcher}} at the top of the navigation, and select **Developer**.
```
Staying consistent with perspective switcher terminology to be consistent with dev console docs.

serenamarie125 commented on quickstarts/quarkus-with-helm.yaml:
```diff
@@ -0,0 +1,138 @@
+        
+        1. In the Helm Chart catalog, click on the **Quarkus** tile.
+        
+            - The side panel displays information about the Quarkus chart
```
this isn't accurate

serenamarie125 commented on quickstarts/quarkus-with-helm.yaml:
```diff
@@ -0,0 +1,138 @@
+        
+            - The side panel displays information about the Quarkus chart
+        
+        1. Click **Install Helm Chart**
```
this isn't in the modal

serenamarie125 commented on quickstarts/quarkus-with-helm.yaml:
```diff
@@ -0,0 +1,138 @@
+        
+        1. Hover over the icon on the bottom left quadrant of the **quarkus** deployment to see the build status in a tooltip.
+        
+        1. Click on the icon for quick access to the build log.  
```
same concern with use of "status icon"

serenamarie125 commented on quickstarts/quarkus-with-helm.yaml:
```diff
@@ -0,0 +1,138 @@
+        
+        1. Click on the icon for quick access to the build log.  
+        
+            - The log stream of the **quarkus-1** build is shown on the **Build Details** page.
```
this isn't accurate.

serenamarie125 commented on quickstarts/quarkus-with-s2i.yaml:
```diff
@@ -0,0 +1,210 @@
+    - description: >-
+        To create a Quarkus application:  
+
+        1. Click on the [perspective switcher]{{highlight qs-perspective-switcher}} at the top of the navigation, and select **Developer**.
```
sticking with perspective switcher to be consistent with the dev console documentation

serenamarie125 commented on quickstarts/quarkus-with-s2i.yaml:
```diff
@@ -0,0 +1,210 @@
+        
+            - You should be able
+        to see the log stream of the **quarkus-quickstarts-1** build on the
+        **Build Details** page. 
```
not accurate.

serenamarie125 commented on quickstarts/quarkus-with-s2i.yaml:
```diff
@@ -0,0 +1,210 @@
+        1. Click the icon for quick access to the build log.  
+        
+            - You should be able
+        to see the log stream of the **quarkus-quickstarts-1** build on the
```
not accurate

serenamarie125 commented on quickstarts/manage-helm-repos.yaml:
```diff
@@ -0,0 +1,79 @@
+
+          1. Click on [Add]{{highlight qs-nav-add}} in the navigation.
+          
+          1. Go to the Helm Chart Catalog
```
I updated this to explain how to navigate to the helm chart repo

serenamarie125 commented:
/retest

beaumorley commented:
just reviewed all the edits. formatting looks great. ux approves.

andrewballantyne commented:
/lgtm

andrewballantyne commented:
/hold cancel

andrewballantyne commented:
/lgtm
